detection-rules/attachment_js_file_execution.yml

```
@@ -0,0 +1,27 @@
+name: "Attachment: File execution via Javascript"
+description: |
+  Javascript contains identifiers or strings that may attempt to execute files.
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound 
+  and any(attachments, .file_type in $file_extensions_common_archives
+      and any(beta.binexplode(.), 
+          any(.scan.javascript.identifiers, strings.ilike(., 'ActiveXObject', 'ShellExecute'))
+          or all(.scan.javascript.strings, strings.ilike(., 'Shell.Application', '*.exe'))
```

```suggestion
          or (length(.scan.javascript.strings) > 0
            and all(.scan.javascript.strings, strings.ilike(., 'Shell.Application', '*.exe')))
```

In order to avoid the vacuous truth caused by .scan.javascript.strings being null, we can encapsulate a length check with the all condition.